src/js/media-store/request-map.ts

```diff
@@ -11,35 +11,18 @@ import {
   updateTracksModeTo,
 } from '../utils/captions.js';
 import { getSubtitleTracks, toggleSubtitleTracks } from './util.js';
+import { StateMediator, StateOwners } from './state-mediator.js';
+import { MediaState } from './media-store.js';
 
-/**
- * @typedef {Omit<
- *  MediaUIEvents,
- *  'REGISTER_MEDIA_STATE_RECEIVER' |
- *  'UNREGISTER_MEDIA_STATE_RECEIVER' |
- *  'MEDIA_SHOW_TEXT_TRACKS_REQUEST' |
- *  'MEDIA_HIDE_TEXT_TRACKS_REQUEST'
- * >} MediaRequestTypes
- */
-
-/**
- * @typedef {import('./state-mediator.js').StateMediator} StateMediator
- */
-
-/**
- * @typedef {import('./state-mediator.js').StateOwners} StateOwners
- */
-
-/** @TODO Duplicate definition (See media-store.js). Consolidate somewhere */
-/**
- * @typedef {{
- *   [K in keyof StateMediator]: ReturnType<StateMediator[K]['get']>
- * } & {
- *   mediaPreviewTime: number;
- *   mediaPreviewImage: string;
- *   mediaPreviewCoords: [number,number,number,number]
- * }} MediaState
- */
+export type MediaUIEventsType =
+  typeof MediaUIEvents[keyof typeof MediaUIEvents];
+export type MediaRequestTypes = Exclude<
```

Comment:
is this correct? Omit to Exclude and uppercast to lowercase

Reply:
Yes this is correct. The `requestMap` below uses the values of `MediaUIEvents` as keys, and `MediaUIEventsType` is a union of all the `MediaUIEvents` values.